models/progresses.js

```diff
@@ -47,9 +48,31 @@ const createProgressDocument = async (progressData) => {
  * @throws {Error} If the userId or taskId is invalid or does not exist.
  **/
 const getProgressDocument = async (queryParams) => {
+  const { dev } = queryParams;
   await assertUserOrTaskExists(queryParams);
   const query = buildQueryToFetchDocs(queryParams);
   const progressDocs = await getProgressDocs(query);
+
+  if (dev) {
+    try {
+      const uniqueUserIds = [...new Set(progressDocs.map((doc) => doc.userId))];
+      const usersData = await Promise.all(uniqueUserIds.map((userId) => fetchUser({ userId })));
+
```

Comment:
While fetching `usersData`, the use of nested `Promise.all` could cause performance issues for large datasets. We should consider batching the user IDs and using an optimized solution.

Reply:
I have used batches approach to fetch userIds